Application/EdFi.Ods.Api/Filters/DataManagementRequestContextFilter.cs

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System;
+using System.Linq;
+using EdFi.Ods.Api.Constants;
+using EdFi.Ods.Common.Models;
+using EdFi.Ods.Common.Models.Resource;
+using EdFi.Ods.Common.Security.Claims;
+using log4net;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace EdFi.Ods.Api.Filters;
+
+/// <summary>
+/// An action filter that inspects the action descriptor's AttributeRouteInfo to locate
+/// the <see cref="Resource" /> associated with the current data management API request.
+/// </summary>
+public class DataManagementRequestContextFilter : IActionFilter
+{
+    private readonly IDataManagementRequestContextProvider _contextProvider;
+
+    private readonly string[] _knownSchemas;
+
+    private readonly ILog _logger = LogManager.GetLogger(typeof(DataManagementRequestContextFilter));
+    private readonly IResourceModelProvider _resourceModelProvider;
+
+    public DataManagementRequestContextFilter(
+        IResourceModelProvider resourceModelProvider,
+        IDataManagementRequestContextProvider contextProvider)
+    {
+        _resourceModelProvider = resourceModelProvider;
+
+        _knownSchemas = _resourceModelProvider.GetResourceModel()
+            .SchemaNameMapProvider.GetSchemaNameMaps()
+            .Select(m => m.UriSegment)
+            .ToArray();
+
+        _contextProvider = contextProvider;
+    }
+
+    public void OnActionExecuting(ActionExecutingContext context)
+    {
+        var attributeRouteInfo = context.ActionDescriptor.AttributeRouteInfo;
+
+        if (attributeRouteInfo != null)
+        {
+            string template = attributeRouteInfo.Template;
+
+            // e.g. data/v3/ed-fi/gradebookEntries
+
+            // Is this a data management route?
+            if (template?.StartsWith(RouteConstants.DataManagementRoutePrefix) ?? false)
+            {
+                var parts = template.Substring(RouteConstants.DataManagementRoutePrefix.Length + 1).Split('/');
+
+                string schema = null, resourceCollection = null;
```

Comment:
Actually, it's the initialized values of `null` that aren't necessary, according to the compiler. They were needed while the code was still being written, but where it ended up, they are not needed. I'll clean it up... with another commit.

Reply:
Yup you're right. I need to get my eyes looked at I guess since the grey text of schema and resourceCollection looked the same as how it highlights the initial value of null being okay to remove.